File: БР1.2/Smirnova_Karina/labs/lab1/src/main/java/org/renting/rentingservice/dto/auth/RegisterRequest.java
package org.renting.rentingservice.dto.auth;

import jakarta.validation.constraints.*;
import lombok.Data;

@Data
public class RegisterRequest {
    @NotBlank
    @Email
    private String email;

    @NotBlank
    @Size(min = 3, max = 50)
    private String username;

    @NotBlank
    @Pattern(regexp = "^[+]?[0-9]+$")
    private String phone;

    @NotBlank
    @Size(min = 6)
    private String password;
}

File: БР1.2/Цатинян Артём/labs/lab1/src/main/kotlin/ru/itmo/restaurantbooking/auth/adapter/rest/dto/RegisterRequest.kt
package ru.itmo.restaurantbooking.auth.adapter.rest.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class RegisterRequest(
    @field:Email
    @field:NotBlank
    val email: String,
    @field:NotBlank
    @field:Size(min = 8, max = 100)
    val password: String,
    @field:NotBlank
    val firstName: String,
    @field:NotBlank
    val lastName: String,
    @field:NotBlank
    val phone: String
)
